# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.h
# Commit: 0c5428075ca4
# Full Hash: 0c5428075ca4aa6b5270350247d43217fa5d0feb
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-07 08:31:23
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r 9e7fe3dbea01 -r 0c5428075ca4 dom/ipc/WindowGlobalParent.h
--- a/dom/ipc/WindowGlobalParent.h	Fri Jun 06 14:38:54 2025 +0000
+++ b/dom/ipc/WindowGlobalParent.h	Fri Jun 06 14:38:54 2025 +0000
@@ -318,19 +318,6 @@
 
   mozilla::ipc::IPCResult RecvReloadWithHttpsOnlyException();
 
-  mozilla::ipc::IPCResult RecvGetIdentityCredential(
-      IdentityCredentialRequestOptions&& aOptions,
-      const CredentialMediationRequirement& aMediationRequirement,
-      bool aHasUserActivation, const GetIdentityCredentialResolver& aResolver);
-  mozilla::ipc::IPCResult RecvDisconnectIdentityCredential(
-      const IdentityCredentialDisconnectOptions& aOptions,
-      const DisconnectIdentityCredentialResolver& aResolver);
-  mozilla::ipc::IPCResult RecvSetLoginStatus(
-      LoginStatus aStatus, const SetLoginStatusResolver& aResolver);
-
-  mozilla::ipc::IPCResult RecvPreventSilentAccess(
-      const PreventSilentAccessResolver& aResolver);
-
   mozilla::ipc::IPCResult RecvGetStorageAccessPermission(
       bool aIncludeIdentityCredential,
       GetStorageAccessPermissionResolver&& aResolve);
@@ -347,6 +334,8 @@
   already_AddRefed<dom::PWebAuthnTransactionParent>
   AllocPWebAuthnTransactionParent();
 
+  already_AddRefed<dom::PWebIdentityParent> AllocPWebIdentityParent();
+
  private:
   WindowGlobalParent(CanonicalBrowsingContext* aBrowsingContext,
                      uint64_t aInnerWindowId, uint64_t aOuterWindowId,